lib/Cleanup3.ml

```diff
@@ -159,6 +159,22 @@ let add_extra_type_to_slice_index =
       | _ -> super#visit_EApp env e es
   end
 
+(*This identifies the decls which should be generated after monomorphism, but is already defined
+ in eurydice_glue.h for implementing the builtin functions. Currently only for arr<T;N> *)
+let is_builtin_lid lid = match lid with
```

Comment:
if you wanted, you could customize lib/krml/Monomorphization.ml to pick readable names in a few specific cases, such as when the type argument if of the form `[T; N]` (and use the hash otherwise)